tests/integ/lock.py

```diff
@@ -15,24 +15,21 @@
 import fcntl
 import os
 import time
+import tempfile
 from contextlib import contextmanager
 
-import tests.integ
-
-LOCK_PATH = os.path.join(tests.integ.DATA_DIR, 'local_mode_lock')
+DEFAULT_LOCK_PATH = os.path.join(tempfile.gettempdir(), 'sagemaker_test_lock')
 
 
 @contextmanager
-def lock():
-    # Since Local Mode uses the same port for serving, we need a lock in order
-    # to allow concurrent test execution.
```

Comment:
I think it'd be good to have an explanation of why these locks are needed somewhere

Reply:
probably at (multiple) points of use. or would a generic "lock for tests that need locks" comment work?

Reply:
maybe a generic comment with specific callouts for the most common reasons (e.g. local mode)?